scripts/mapo

```diff
@@ -0,0 +1,231 @@
+    )
+    optimizers_args_group.add_argument(
+        "--critic-delay",
+        type=int,
+        default=1,
+        help="The number of traing steps before updating the critic " "(default=2).",
+    )
+
+    exec_args_group = parser.add_argument_group(">> Execution")
+    exec_args_group.add_argument(
+        "--timesteps-total",
+        type=int,
+        default=1000,
+        help="The total number of timesteps (default=1000).",
+    )
+    exec_args_group.add_argument(
+        "--sample-batch-size",
+        type=int,
+        default=10,
+        help="The size of the batch collected by workers (default=10).",
+    )
+    exec_args_group.add_argument(
+        "--train-batch-size",
+        type=int,
+        default=100,
+        help="The size of the batch for training (default=100).",
+    )
+
+    resources_args_group = parser.add_argument_group(">> Resources")
+    resources_args_group.add_argument(
+        "--num-workers",
+        type=int,
+        default=0,
+        help="The number of actors used for parallelism (default=0).",
+    )
+    resources_args_group.add_argument(
+        "--num-gpus",
+        type=float,
+        default=0.0,
+        help="The number of GPUs to allocate to the driver (default=0.0).",
+    )
+    resources_args_group.add_argument(
+        "--num-cpus",
+        type=int,
+        default=1,
+        help="The number of CPUs to allocate to the driver (default=1).",
+    )
+
+    exp_args_group = parser.add_argument_group(">> Experiment")
+    exp_args_group.add_argument(
+        "--num-samples",
+        type=int,
+        default=1,
+        help="The number of times to sample from the hyperparameter space "
+        "(default=1).",
+    )
+    exp_args_group.add_argument("--name", type=str, help="The name of the experiment.")
+
+    return parser.parse_args()
+
+
+def init():
+    """Initialize ray framework."""
+    ray.init()
+
+
+def register():
+    """Register all agents and custom environtments."""
+    register_all_agents()
+    register_all_environments()
+
+
+def run(args):
+    """Run experiments with configuration given by command-line arguments."""
+    trainer = args.run
+    experiment = args.name
+
+    analysis = tune.run(
+        trainer,
+        name=experiment,
+        verbose=args.verbose,
+        stop={"timesteps_total": args.timesteps_total},
+        # resources_per_trial={
+        #     "cpu": args.num_cpus,
+        #     "gpu": args.num_gpus,
+        # },
+        config={
+            "env": tune.grid_search(args.env),
+            "num_workers": args.num_workers,
+            "model": {
+                "custom_model": "mapo_model",
+                "custom_options": {
+                    "actor": args.config_actor_net,
+                    "critic": args.config_critic_net,
+                    "dynamics": args.config_dynamics_net,
+                },
+            },
+        },
```

Comment:
Just to be clear: are we purposefully ignoring some command line arguments such as delays and sample batch sizes?

Reply:
I forgot to add those. Just did it in this follow-up commit b377b99

Reply:
But again, there's still an awful lot of arguments to add... this is just a scaffold... also not entirely confident there isn't a better way of passing args from CLI to the trainer's config dict... but let's no dwell too much with details here..

Reply:
BTW, I've just noticed that for ``trainer=MAPO`` passing the argument `critic_delay` in config causes an error:

```bash
$ mapo --run=MAPO --env=Navigation-v0 --config-actor scripts/actor.json --config-critic scripts/critic.json --config-dynamics scripts/dynamics.json --timesteps-total 1000
```

```bash
Traceback (most recent call last):
  File "/home/tbueno/Projects/model-based-reinforcement-learning/lib/python3.7/site-packages/ray/tune/trial_runner.py", line 498, in _process_trial
    result = self.trial_executor.fetch_result(trial)
  File "/home/tbueno/Projects/model-based-reinforcement-learning/lib/python3.7/site-packages/ray/tune/ray_trial_executor.py", line 342, in fetch_result
    result = ray.get(trial_future[0])
  File "/home/tbueno/Projects/model-based-reinforcement-learning/lib/python3.7/site-packages/ray/worker.py", line 2247, in get
    raise value
ray.exceptions.RayTaskError: ray_MAPO:train() (pid=24875, host=system76-ubuntu)
  File "/home/tbueno/Projects/model-based-reinforcement-learning/lib/python3.7/site-packages/ray/rllib/agents/trainer_template.py", line 87, in __init__
    Trainer.__init__(self, config, env, logger_creator)
  File "/home/tbueno/Projects/model-based-reinforcement-learning/lib/python3.7/site-packages/ray/rllib/agents/trainer.py", line 328, in __init__
    Trainable.__init__(self, config, logger_creator)
  File "/home/tbueno/Projects/model-based-reinforcement-learning/lib/python3.7/site-packages/ray/tune/trainable.py", line 99, in __init__
    self._setup(copy.deepcopy(self.config))
  File "/home/tbueno/Projects/model-based-reinforcement-learning/lib/python3.7/site-packages/ray/rllib/agents/trainer.py", line 429, in _setup
    self._allow_unknown_subkeys)
  File "/home/tbueno/Projects/model-based-reinforcement-learning/lib/python3.7/site-packages/ray/tune/util.py", line 172, in deep_update
    raise Exception("Unknown config parameter `{}` ".format(k))
Exception: Unknown config parameter `critic_delay` 
```
Is this related to the fact in `mapo.py` trainer module there is no `critic_delay` in the `DEFAULT_CONFIG` ?

Reply:
Probably the case, I think I fixed this in my other PR (#50)

Reply:
In which commit / file?

Reply:
This one 
39d4d13

Reply:
Thanks!